presto-spi/src/main/java/com/facebook/presto/spi/statistics/PlanStatisticsWithSourceInfo.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi.statistics;
+
+import com.facebook.presto.spi.plan.PlanNodeId;
+
+import java.util.Objects;
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+
+public class PlanStatisticsWithSourceInfo
+{
+    private final PlanNodeId id;
```

Comment:
why do you need the plan node id?

Reply:
It is useful for logging. Alternative will be to store as `MultiMap<PlanNodeId, PlanStatisticsWithSourceInfo>`

Reply:
ah okay, it looked like it was used just in a map with PlanNodeWithHash, which already contains the id, so it didn't seem necessary to have it here.